crates/doc/src/shape/widen.rs

```diff
@@ -32,14 +32,14 @@ impl StringShape {
             }
         }
 
-        if self.min_length as usize > val.len() {
-            self.min_length = length_bounds(val.len()).0;
+        if self.min_length as usize > val.chars().count() {
```

Comment:
nit: `val.chars().count()` is called multiple times (min and max cases). Hoist into variable?